other/blacklistipchecker.py

```
@@ -0,0 +1,101 @@
+######################################
+# This CLI script checks if entered ips are or aren`t blacklisted on different lists
+# After configuration it can even send an email report
+# You need to get API key on blacklistchecker to make it work (it`s free 30 IPs per month)
+######################################
+
+
+import requests
+import sys
+from IPy import IP
+import smtplib
+from email.message import EmailMessage
+import json
+
+API_key = ''
+subject = ''
+from = ''
+to = ''
+smtp_user = ''
+smtp_passw = ''
+smtp_serv = ''
+smtp_port = 587
+
+def test_ip(ip):
+    detects = []
+    result = requests.get('https://api.blacklistchecker.com/check/'+ip, auth=(API_Key,''))
+    result_dec = json.loads(result.content)
+    print(result_dec)
+    if result_dec.get('statusCode'):
+        res="limit exceeded"
+        detects = "!"
+        return(res, detects)
+    elif result_dec.get('detections') != 0:
+        res = ip
+        blsts = result_dec.get('blacklists')
+        for n in blsts:
+            if n.get('detected') == True:
+                detects.append(n.get('name'))
+        return(res, detects)
+    else:
+        return()
+
+res_txt = 'IP blacklists check returned next results:\n'
+send_param = 0
+if len(sys.argv) > 1:
+    for x in sys.argv[1:]:
+        if x in '-y-Y':
+            send_param = 1
+        else:
+            try:
+                IP(x)
+            except:
```

OUCH!!  A bare except!!